--- a/mpdecimal32.h	2010-11-04 14:48:30.000000000 +0100
+++ b/mpdecimal32.h	2010-11-04 16:03:35.000000000 +0100
@@ -64,8 +64,9 @@
 typedef uint64_t mpd_uuint_t; /* double width unsigned mod type */
 #endif
 
-#define MPD_SIZE_MAX SIZE_MAX
-typedef size_t mpd_size_t; /* unsigned size type */
+/* enable CONFIG_32+ANSI on 64-bit platforms without resorting to -m32 */
+#define MPD_SIZE_MAX UINT32_MAX
+typedef uint32_t mpd_size_t; /* unsigned size type */
 
 /* type for dec->len, dec->exp, ctx->prec */
 #define MPD_SSIZE_MAX INT32_MAX
@@ -91,8 +92,8 @@
 #define MPD_MAXIMPORT 94444445L      /* ceil((2*MPD_MAX_PREC)/MPD_RDIGITS) */
 
 
-#if MPD_SIZE_MAX != MPD_UINT_MAX
-  #error "unsupported platform: need mpd_size_t == mpd_uint_t"
+#if SIZE_MAX < MPD_SIZE_MAX
+  #error "unsupported platform: need size_t >= mpd_size_t"
 #endif
 
 
